yo1k/tic_tac_toe/test/core/test_game.py

```diff
@@ -0,0 +1,266 @@
+        for i in list_of_cell_and_win_board:
+            cell, win_board = i
+            state = _new_state(board=g.Board(win_board))
+            with self.subTest(win_board=state.board.cells):
+                self.assertTrue(g.Logic._win_condition(state.board, cell))
+
+    def test_not_win_condition(self):
+        list_of_cell_and_not_win_board = [
+            (g.Cell(1, 0),
+             [[g.Mark.X, g.Mark.X, None], [g.Mark.X, None, None], [None, None, None]]),
+            (g.Cell(0, 0),
+             [[g.Mark.X, None, None], [None, None, g.Mark.X], [None, g.Mark.X, None]]),
+            (g.Cell(0, 1),
+             [[None, g.Mark.X, None], [None, None, g.Mark.X], [None, g.Mark.X, None]]),
+            (g.Cell(0, 2),
+             [[g.Mark.X, g.Mark.O, g.Mark.X], [None, None, None], [None, None, None]])]
+        for i in list_of_cell_and_not_win_board:
+            cell, not_win_board = i
+            state = _new_state(board=g.Board(not_win_board))
+            with self.subTest(not_win_board=state.board.cells):
+                self.assertFalse(g.Logic._win_condition(state.board, cell))
+
+    def test_advance_win_action(self):
+        previous_game_state_board = [
+            [g.Mark.X, None, None], [g.Mark.O, g.Mark.X, g.Mark.O], [None, None, None]]
+        state = _new_state(
+            board=g.Board(previous_game_state_board),
+            step=4)
+        act_queue_px = ListActionQueue([g.Action.new_occupy(g.Cell(2, 2))])
+        g.Logic([act_queue_px, []]).advance(state)
+        next_game_state_board = [
+            [g.Mark.X, None, None], [g.Mark.O, g.Mark.X, g.Mark.O], [None, None, g.Mark.X]]
+        self.assertEqual(state.board.cells, next_game_state_board)
+        self.assertEqual(state.step, 4)
+        self.assertEqual(state.round, 0)
+        self.assertEqual(state.phase, g.Phase.OUTROUND)
+        self.assertEqual(state.players[0].wins, 1)
+        self.assertEqual(state.players[1].wins, 0)
+
+    def test_advance_draw_action(self):
+        previous_game_state_board = [
+            [g.Mark.O, g.Mark.X, g.Mark.O],
+            [g.Mark.O, g.Mark.X, g.Mark.X],
+            [g.Mark.X, g.Mark.O, None]]
+        state = _new_state(
+            board=g.Board(previous_game_state_board),
+            step=8)
+        act_queue_px = ListActionQueue([g.Action.new_occupy(g.Cell(2, 2))])
+        g.Logic([act_queue_px, []]).advance(state)
+        next_game_state_board = [
+            [g.Mark.O, g.Mark.X, g.Mark.O],
+            [g.Mark.O, g.Mark.X, g.Mark.X],
+            [g.Mark.X, g.Mark.O, g.Mark.X]]
+        self.assertEqual(state.board.cells, next_game_state_board)
+        self.assertEqual(state.step, 8)
+        self.assertEqual(state.round, 0)
+        self.assertEqual(state.phase, g.Phase.OUTROUND)
+        self.assertEqual(state.players[0].wins, 0)
+        self.assertEqual(state.players[1].wins, 0)
+
+    def test_advance_next_round_action_outround(self):
```

Comment:
I can't read this. Since underscores are used to separate words, maybe two underscores could be used to separate logical parts of a test name. Though, I have doubts that such an approach would help this name. We need to talk about the names of all tests.

Reply:
1
2

Reply:
Done.
I have changed names of all test methods using double underscore.